model/lib/oidc/token_oidc.dart

```diff
@@ -27,6 +28,38 @@ class TokenOIDC with EquatableMixin {
 
   Map<String, dynamic> toJson() => _$TokenOIDCToJson(this);
 
+  factory TokenOIDC.fromUri(String uriString) {
+    final uri = Uri.parse(uriString);
+    final queryParams = uri.queryParameters;
+
+    final accessToken = queryParams['access_token'];
+    if (accessToken == null || accessToken.isEmpty) {
+      throw AccessTokenIsNullException();
+    }
+
+    final refreshToken = queryParams['refresh_token'];
+    if (refreshToken == null || refreshToken.isEmpty) {
+      throw RefreshTokenIsNullException();
+    }
+
+    final idToken = queryParams['id_token'];
+    if (idToken == null || idToken.isEmpty) {
+      throw TokenIdIsNullException();
+    }
+
+    final expiresIn = queryParams['expires_in'];
+    if (expiresIn == null || expiresIn.isEmpty) {
+      throw ExpiresTimeIsNullException();
+    }
```

Comment:
No need to verify this, these are nullable fields which depend on scope of `token` request.

Reply:
done